### PR TITLE
Release 4 3 modularised

### PR DESCRIPTION
**Pull request by seaders**
Main bits and pieces is purely adding new files, moving some of those files / logic of those files into SwrveSDKCommon.  It's broken down into a few specific commits.

After this PR, I suggest we also standardising the "Sdk" folder and moving it into a "SwrveSDK" folder that contains the CocoaPod related files. (this isn't necessary, but is more a cleanup operation)

### CODE REVIEW

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#import <UIKit/UIKit.h>
+
+/*! Swrve SDK shared protocol (interface) definition */
+@protocol ISwrveCommon <NSObject>
+
+-(NSData*) getCampaignData:(int)category;
+-(void) sendQueuedEvents;
+-(int) eventInternal:(NSString*)eventName payload:(NSDictionary*)eventPayload triggerCallback:(bool)triggerCallback;
+-(int) userUpdate:(NSDictionary*)attributes;
+-(void) setLocationVersion:(NSString*)locationVersion;
```
Comment (Sergio-Mira) [inline SwrveSDKCommon/Common/SwrveCommon.h]: Isn't the locationVersions set by the SDK internally? Why expose?

Comment (Sergio-Mira) [inline SwrveSDKCommon/Common/SwrveCommon.h]: Wop confused with conversation version, never mind

File: SwrveSDKCommon/SwrveSDKCommon.podspec [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Pod::Spec.new do |s|
+  s.name             = "SwrveSDKCommon"
+  s.version          = "4.2.0"
```
Comment (Sergio-Mira) [inline SwrveSDKCommon/SwrveSDKCommon.podspec:3]: 4.3

**Comment by Sergio-Mira** [comment]
:rocket:

File: Sdk/Track/Swrve.h [hunk 3]
```diff
@@ -325,7 +319,7 @@ typedef void (^SwrveResourcesUpdatedListener) ();
 @end
 
 /*! Swrve SDK main class. */
-@interface Swrve : NSObject<SwrveSignatureErrorListener>
+@interface Swrve : NSObject<ISwrveCommon, SwrveSignatureErrorListener>
```
Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.h:322]: ISwrveCommon?

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+-(BOOL) processPermissionRequest:(NSString*)action;
+
+@end
+
+@interface SwrveCommon : NSObject
+
++(id<ISwrveCommon>) getSwrveCommon;
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:17]: getInstance?

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    // Delay between flushing events and refreshing campaign/user resources
+    SWRVE_DEFAULT_CAMPAIGN_RESOURCES_FLUSH_REFRESH_DELAY = 5000,
+};
+
+//
+//#ifndef DEBUG
+//
+//#define DEBUG
+//
+//#endif
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:81]: Are these comments valid? Remove if not.

File: SwrveSDKCommon/Common/SwrveCommon.h [hunk 6]
```diff
@@ -0,0 +1,81 @@
+#import <UIKit/UIKit.h>
+
+/*! Swrve SDK shared protocol (interface) definition */
+@protocol ISwrveCommon <NSObject>
+
+-(NSData*) getCampaignData:(int)category;
+-(void) sendQueuedEvents;
+-(int) eventInternal:(NSString*)eventName payload:(NSDictionary*)eventPayload triggerCallback:(bool)triggerCallback;
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:8]: Is this exposed publicly now?

File: SwrveSDKCommon/Common/SwrveCommon.h (hunk 5 above)
Comment (seaders) [inline SwrveSDKCommon/Common/SwrveCommon.h:81]: I basically just pulled these from Swrve.h.

https://github.com/Swrve/swrve-ios-sdk/blob/38d6fa96af870afcff62f0d1d4b9772180060aab/Sdk/Track/Swrve.m#L43

File: Sdk/Track/Swrve.m [hunk 7]
```diff
@@ -1234,6 +1202,17 @@ - (void) checkForCampaignAndResourcesUpdates:(NSTimer*)timer
     }
 }
 
+-(NSData*) getCampaignData:(int)category {
```
Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.m:1205]: I can't see where this is used? Is it for unity?

Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.m:1205]: If it is to be used, then rename to campaignDataForCategory??

File: SwrveSDKCommon/Common/SwrveCommon.h (hunk 4 above)
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:17]: In the regular sdk we use this syntax:
Swrve *swrve = [Swrve sharedInstance];

Should we be doing similar syntax here also?
[SwrveCommon sharedInstance]

File: SwrveSDKCommon/Common/SwrveCommon.h (hunk 5 above)
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:81]: They probably don't make sense anymore......or at least I don't see what they mean.

File: SwrveSDKCommon/LICENSE [hunk 8]
```diff
@@ -0,0 +1,190 @@
+Copyright 2010-2015 Swrve Mobile Inc, or its licensors
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/LICENSE:1]: 2016

File: Sdk/Track/Swrve.h (hunk 3 above)
Comment (seaders) [inline Sdk/Track/Swrve.h:322]: Paging @adam-govan & @Sergio-Mira too.

So `SwrveCommonDelegate`?  Just `SwrveDelegate`?

File: SwrveSDKCommon/Common/SwrveCommon.h (hunk 5 above)
Comment (seaders) [inline SwrveSDKCommon/Common/SwrveCommon.h:81]: I certainly didn't...  So just remove?

Comment (dominicmarmionswrve) [inline SwrveSDKCommon/Common/SwrveCommon.h:81]: Might aswell

**Pull request merged**